src/components/proofs/Proof.tsx

```diff
@@ -54,7 +53,6 @@ function useProofContent(
   color: 'text-accent' | 'text-secondary' | 'text-tertiary'
   content: JSX.Element | null
 } {
-  const { account } = useSnapshot(WalletStore)
```

Comment:
This change seems harmful. We should be using snapshots while rendering, not proxy states.

Reply:
Changed it

Reply:
If I return this line back, it will throw an error on switching accounts. This line causes that effect for the list of proofs

Reply:
You need to find a different solution, you **should not** use proxy state inside components, only the snapshots. The fact that snapshots break the UI tells us that there is an underlying problem that needs to be solved, not patched like this.